engine/model.py

```diff
@@ -324,19 +326,34 @@ def set_player_facing(self, facing_x: int, facing_y: int) -> None:
         self.player_sprite.facing_y = facing_y
 
     def activate(self) -> None:
-        """Activates whatever is in front of the player."""
+        """
+        Activates whatever is in front of the player,
+        by checking for collision with a hitbox_sprite.
+        """
 
         # Do a little bit of math to figure out where to place the hitbox.
         facing = pmath.Vec2(self.player_sprite.facing_x, self.player_sprite.facing_y)
 
         hitbox_center = facing.normalize().scale(HITBOX_DISTANCE)
 
-        objects = arcade.get_sprites_at_point(
-            (
-                int(hitbox_center.x + self.player_sprite.center_x),
-                int(hitbox_center.y + self.player_sprite.center_y),
-            ),
-            self.scene.get_sprite_list(SCRIPTED_OBJECTS),
+        hitbox_corners = [
+            (-HITBOX_WIDTH / 2, -HITBOX_HEIGHT / 2),
+            (HITBOX_WIDTH / 2, -HITBOX_HEIGHT / 2),
+            (HITBOX_WIDTH / 2, HITBOX_HEIGHT / 2),
+            (-HITBOX_WIDTH / 2, HITBOX_HEIGHT / 2),
+        ]
+        hitbox_sprite = arcade.Sprite(
+            center_x=hitbox_center.x + self.player_sprite.center_x,
+            center_y=hitbox_center.y + self.player_sprite.center_y,
+        )
+        hitbox_sprite.set_hit_box(
+            [
+                (corner_x + hitbox_center.x, corner_y + hitbox_center.y)
+                for corner_x, corner_y in hitbox_corners
+            ]
+        )
+        objects = arcade.check_for_collision_with_list(
+            hitbox_sprite, self.scene.get_sprite_list(SCRIPTED_OBJECTS)
```

Comment:
Put a comma at the end of the line so that Black formats this correctly. See [the README](https://github.com/robbrit/ossu-game-project-1#function-breaks).

Reply:
My bad.